pages/ideaDetails/[id].js

```diff
@@ -0,0 +1,95 @@
+import Header from "@/components/Header";
+import Image from "next/image";
+import Link from "next/link";
+import styled from "styled-components";
+
+import { v4 as uuidv4 } from "uuid";
+import { ideas as defaultIdeas } from "@/lib/db";
+import { useState } from "react";
+import { useRouter } from "next/router.js";
+
+const StyledArticle = styled.article`
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1rem;
+  list-style: none;
+  padding: 0;
+  margin-top: 1rem;
+  flex-direction: column;
+`;
+
+const StyledImage = styled(Image)`
+  border-radius: 0.5rem;
+`;
+
+const StyledContainer = styled.div`
+  display: flex;
+  gap: 1rem;
+`;
+
+const Instruction = styled.ol`
+  padding-left: 1rem;
+`;
+
+const Hashtags = styled.ul`
+  list-style: none;
+  display: flex;
+  gap: 0.5rem;
+  flex-wrap: wrap;
+`;
+
+const Items = styled.ul`
+  list-style: none;
+  border: 1px solid black;
+  border-radius: 0.5rem;
+  padding: 0.5rem;
+`;
+
+export default function IdeaDetails() {
+  const [ideas, setIdeas] = useState(defaultIdeas);
+  const router = useRouter();
+
+  const { id } = router.query;
+
+  const ideaDetails = ideas.find((idea) => idea.id === id);
+
+  if (!ideaDetails) {
+    return <h2>Loading...</h2>;
+  }
+
+  const { instructions, items, hashtags } = ideaDetails;
+
+  return (
+    <>
+      <Header />
```

Comment:
For now, this is alright. In the future I'd suggest moving the `<Header>` and potentially `<Navigation>` to its own `<Layout>` component that you'd import in your `_app.js` and wrap it around your entire app to make it available everywhere.

Reply:
We will implement this in the future, thanks!